diff --git a/node_modules/react-login-modal-sm/dist/index b/node_modules/react-login-modal-sm/dist/index
index bf548b5..c3654ca 100644
--- a/node_modules/react-login-modal-sm/dist/index
+++ b/node_modules/react-login-modal-sm/dist/index
@@ -534,7 +534,7 @@ var LoginModal = function (_Component) {
     }
 
     return _ret = (_temp = (_this = possibleConstructorReturn(this, (_ref = LoginModal.__proto__ || Object.getPrototypeOf(LoginModal)).call.apply(_ref, [this].concat(args))), _this), _this.state = {
-      subpageShown: 0
+      subpageShown: 1
     }, _this.switchSubpage = function (to) {
       return function (e) {
         _this.setState({ subpageShown: to });
diff --git a/node_modules/react-login-modal-sm/dist/index.js b/node_modules/react-login-modal-sm/dist/index.js
index 098817b..af3e530 100644
--- a/node_modules/react-login-modal-sm/dist/index.js
+++ b/node_modules/react-login-modal-sm/dist/index.js
@@ -29,6 +29,7 @@ var emailRegex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((
 var usernameRegex = /^[a-zA-Z0-9_-]{5,}/;
 var passwordRegex = /^[a-zA-Z0-9_@!#()]{8,}/;
 
+
 var focusStyle = function focusStyle(focus) {
   return focus ? { backgroundColor: "#fff" } : { backgroundColor: "#e6e7f2" };
 };
@@ -319,7 +320,7 @@ var SignupForm = function (_Component) {
       var emailRegex$$1 = new RegExp(_this.props.emailRegex);
       var usernameRegex$$1 = new RegExp(_this.props.usernameRegex);
       var passwordRegex$$1 = new RegExp(_this.props.passwordRegex);
-      return emailRegex$$1.test(_this.state.email) && usernameRegex$$1.test(_this.state.username) && passwordRegex$$1.test(_this.state.password);
+      return emailRegex$$1.test(_this.state.email) && passwordRegex$$1.test(_this.state.password);
     }, _this.onSignupFacebook = function () {
       if (_this.props.onSignupFacebook === undefined) console.warn("No function found: Pass a function as props; onSignupFacebook");else _this.props.onSignupFacebook();
     }, _this.onSignupGoogle = function () {
@@ -534,7 +535,7 @@ var LoginModal = function (_Component) {
     }
 
     return _ret = (_temp = (_this = possibleConstructorReturn(this, (_ref = LoginModal.__proto__ || Object.getPrototypeOf(LoginModal)).call.apply(_ref, [this].concat(args))), _this), _this.state = {
-      subpageShown: 0
+      subpageShown: 1
     }, _this.switchSubpage = function (to) {
       return function (e) {
         _this.setState({ subpageShown: to });
@@ -544,7 +545,7 @@ var LoginModal = function (_Component) {
     }, _this.emailRegex = function () {
       return _this.props.emailRegex !== undefined ? _this.props.emailRegex : emailRegex;
     }, _this.usernameRegex = function () {
-      return _this.props.usernameRegex !== undefined ? _this.props.usernameRegex : usernameRegex;
+      return true;
     }, _this.passwordRegex = function () {
       return _this.props.passwordRegex !== undefined ? _this.props.passwordRegex : passwordRegex;
     }, _this.closeModal = function () {
