diff --git a/node_modules/react-login-modal-sm/dist/index b/node_modules/react-login-modal-sm/dist/index
index bf548b5..c3654ca 100644
--- a/node_modules/react-login-modal-sm/dist/index
+++ b/node_modules/react-login-modal-sm/dist/index
@@ -534,7 +534,7 @@ var LoginModal = function (_Component) {
     }
 
     return _ret = (_temp = (_this = possibleConstructorReturn(this, (_ref = LoginModal.__proto__ || Object.getPrototypeOf(LoginModal)).call.apply(_ref, [this].concat(args))), _this), _this.state = {
-      subpageShown: 0
+      subpageShown: 1
     }, _this.switchSubpage = function (to) {
       return function (e) {
         _this.setState({ subpageShown: to });
diff --git a/node_modules/react-login-modal-sm/dist/index.es.js b/node_modules/react-login-modal-sm/dist/index.es.js
index 707c685..1a4e095 100644
--- a/node_modules/react-login-modal-sm/dist/index.es.js
+++ b/node_modules/react-login-modal-sm/dist/index.es.js
@@ -529,7 +529,7 @@ var LoginModal = function (_Component) {
     }
 
     return _ret = (_temp = (_this = possibleConstructorReturn(this, (_ref = LoginModal.__proto__ || Object.getPrototypeOf(LoginModal)).call.apply(_ref, [this].concat(args))), _this), _this.state = {
-      subpageShown: 0
+      subpageShown: 1
     }, _this.switchSubpage = function (to) {
       return function (e) {
         _this.setState({ subpageShown: to });
diff --git a/node_modules/react-login-modal-sm/dist/index.js b/node_modules/react-login-modal-sm/dist/index.js
index 098817b..a484e67 100644
--- a/node_modules/react-login-modal-sm/dist/index.js
+++ b/node_modules/react-login-modal-sm/dist/index.js
@@ -534,7 +534,7 @@ var LoginModal = function (_Component) {
     }
 
     return _ret = (_temp = (_this = possibleConstructorReturn(this, (_ref = LoginModal.__proto__ || Object.getPrototypeOf(LoginModal)).call.apply(_ref, [this].concat(args))), _this), _this.state = {
-      subpageShown: 0
+      subpageShown: 1
     }, _this.switchSubpage = function (to) {
       return function (e) {
         _this.setState({ subpageShown: to });
